[spack-build-out.txt.gz](https://github.com/spack/spack/files/6568596/spack-build-out.txt.gz)
But installing `tix` fails:

```console
==> Installing tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz
==> No binary for tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz found: installing from source
==> Fetching https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/_source-cache/archive/56/562f040ff7657e10b5cffc2c41935f1a53c6402eb3d5f3189113d734fd6c03cb.tar.gz
#=#=-  #       #                                                                                                                                                                                                                                               #=O=#     #        #                                                                                                                                                                                                                                           -#O=- #      #          #       ######################################################################################################################################################################################################################################################### 100.0%
==> Fetching https://raw.githubusercontent.com/macports/macports-ports/master/x11/tix/files/panic.patch
######################################################################################################################################################################################################################################################### 100.0%
==> Fetching https://raw.githubusercontent.com/macports/macports-ports/master/x11/tix/files/patch-generic-tixGrSort.c.diff
######################################################################################################################################################################################################################################################### 100.0%
==> Fetching https://raw.githubusercontent.com/macports/macports-ports/master/x11/tix/files/implicit.patch
######################################################################################################################################################################################################################################################### 100.0%
==> Applied patch https://raw.githubusercontent.com/macports/macports-ports/master/x11/tix/files/panic.patch
==> Applied patch https://raw.githubusercontent.com/macports/macports-ports/master/x11/tix/files/patch-generic-tixGrSort.c.diff
==> Applied patch https://raw.githubusercontent.com/macports/macports-ports/master/x11/tix/files/implicit.patch
==> tix: Executing phase: 'autoreconf'
==> tix: Executing phase: 'configure'
==> tix: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j8'

8 errors found in build log:
     94     /home/sschmitt/spack-tcl/lib/spack/env/gcc/gcc -DPACKAGE_NAME=\"Tix\" -DPACKAGE_TARNAME=\"tix\" -DPACKAGE_VERSION=\"8.4.3\" -DPACKAGE_STRING=\"Tix\ 8.4.3\" -DPACKAGE_BUGREPORT=\"\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -
            DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_LIMITS_H=1 -DHAVE_SYS_PARAM_H=1 -DUSE_THREAD_ALLOC=1 -D_REENTRANT=1 -D_THREAD_SAFE=1 -DTCL_THREADS=1 -D_LARGEFILE64_SOURCE=1 -DTCL_WIDE_INT_I
            S_LONG=1 -DUSE_TCL_STUBS=1 -DUSE_TK_STUBS=1   -I. -I"/tmp/sschmitt/spack-stage/spack-stage-tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz/spack-src/generic" -I. -I"/tmp/sschmitt/spack-stage/spack-stage-tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz/spack-src/unix" -
            I"/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tcl-8.6.11-jgl7fpo5wz37kcve7pskehab2tbnr4h7/share/tcl/src/generic" -I"/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tcl-8.6.11-jgl7fpo5wz37kcve7pskehab2tbnr4h7/
            share/tcl/src/unix" -I"/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tk-8.6.11-3pygupoqtd3fixjimjfyu34o6zesxrgu/share/tk/src/generic" -I"/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tk-8.6.11-3pygupoqtd3fixj
            imjfyu34o6zesxrgu/share/tk/src/unix" -I/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/libx11-1.7.0-3ukzesrsqcspevi4a6svf3ynxalrwpx7/include    -pipe -O2 -fomit-frame-pointer -Wall -Wno-implicit-int -fPIC  -c `echo /tmp/sschmitt/spack-
            stage/spack-stage-tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz/spack-src/generic/tixInit.c`
     95     /home/sschmitt/spack-tcl/lib/spack/env/gcc/gcc -DPACKAGE_NAME=\"Tix\" -DPACKAGE_TARNAME=\"tix\" -DPACKAGE_VERSION=\"8.4.3\" -DPACKAGE_STRING=\"Tix\ 8.4.3\" -DPACKAGE_BUGREPORT=\"\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -
            DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_LIMITS_H=1 -DHAVE_SYS_PARAM_H=1 -DUSE_THREAD_ALLOC=1 -D_REENTRANT=1 -D_THREAD_SAFE=1 -DTCL_THREADS=1 -D_LARGEFILE64_SOURCE=1 -DTCL_WIDE_INT_I
            S_LONG=1 -DUSE_TCL_STUBS=1 -DUSE_TK_STUBS=1   -I. -I"/tmp/sschmitt/spack-stage/spack-stage-tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz/spack-src/generic" -I. -I"/tmp/sschmitt/spack-stage/spack-stage-tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz/spack-src/unix" -
            I"/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tcl-8.6.11-jgl7fpo5wz37kcve7pskehab2tbnr4h7/share/tcl/src/generic" -I"/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tcl-8.6.11-jgl7fpo5wz37kcve7pskehab2tbnr4h7/
            share/tcl/src/unix" -I"/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tk-8.6.11-3pygupoqtd3fixjimjfyu34o6zesxrgu/share/tk/src/generic" -I"/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tk-8.6.11-3pygupoqtd3fixj
            imjfyu34o6zesxrgu/share/tk/src/unix" -I/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/libx11-1.7.0-3ukzesrsqcspevi4a6svf3ynxalrwpx7/include    -pipe -O2 -fomit-frame-pointer -Wall -Wno-implicit-int -fPIC  -c `echo /tmp/sschmitt/spack-
            stage/spack-stage-tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz/spack-src/generic/tixList.c`
     96     /home/sschmitt/spack-tcl/lib/spack/env/gcc/gcc -DPACKAGE_NAME=\"Tix\" -DPACKAGE_TARNAME=\"tix\" -DPACKAGE_VERSION=\"8.4.3\" -DPACKAGE_STRING=\"Tix\ 8.4.3\" -DPACKAGE_BUGREPORT=\"\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -
            DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_LIMITS_H=1 -DHAVE_SYS_PARAM_H=1 -DUSE_THREAD_ALLOC=1 -D_REENTRANT=1 -D_THREAD_SAFE=1 -DTCL_THREADS=1 -D_LARGEFILE64_SOURCE=1 -DTCL_WIDE_INT_I
            S_LONG=1 -DUSE_TCL_STUBS=1 -DUSE_TK_STUBS=1   -I. -I"/tmp/sschmitt/spack-stage/spack-stage-tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz/spack-src/generic" -I. -I"/tmp/sschmitt/spack-stage/spack-stage-tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz/spack-src/unix" -
            I"/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tcl-8.6.11-jgl7fpo5wz37kcve7pskehab2tbnr4h7/share/tcl/src/generic" -I"/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tcl-8.6.11-jgl7fpo5wz37kcve7pskehab2tbnr4h7/
            share/tcl/src/unix" -I"/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tk-8.6.11-3pygupoqtd3fixjimjfyu34o6zesxrgu/share/tk/src/generic" -I"/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tk-8.6.11-3pygupoqtd3fixj
            imjfyu34o6zesxrgu/share/tk/src/unix" -I/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/libx11-1.7.0-3ukzesrsqcspevi4a6svf3ynxalrwpx7/include    -pipe -O2 -fomit-frame-pointer -Wall -Wno-implicit-int -fPIC  -c `echo /tmp/sschmitt/spack-
            stage/spack-stage-tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz/spack-src/generic/tixMethod.c`
     97     In file included from /home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tcl-8.6.11-jgl7fpo5wz37kcve7pskehab2tbnr4h7/share/tcl/src/generic/tcl.h:2441,
     98                      from /tmp/sschmitt/spack-stage/spack-stage-tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz/spack-src/generic/tixPort.h:22,
     99                      from /tmp/sschmitt/spack-stage/spack-stage-tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz/spack-src/generic/tixGeometry.c:16:
  >> 100    /home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tcl-8.6.11-jgl7fpo5wz37kcve7pskehab2tbnr4h7/share/tcl/src/generic/tclDecls.h:2542:14: error: expected ')' before '->' token
     101     2542 |  (tclStubsPtr->tcl_Panic) /* 2 */
```

--- a/var/spack/repos/builtin/packages/tcl/package.py
+++ b/var/spack/repos/builtin/packages/tcl/package.py
@@ -83,7 +83,11 @@ class Tcl(AutotoolsPackage, SourceforgePackage):
         Returns:
             Executable: the tclsh command
         """
-        return Executable(os.path.realpath(self.prefix.bin.tclsh))
+        # Although we symlink tclshX.Y to tclsh, we also need to support external
+        # installations that may not have this symlink, or may have multiple versions
+        # of Tcl installed in the same directory.
+        return Executable(os.path.realpath(self.prefix.bin.join(
+            'tclsh{0}'.format(self.version.up_to(2)))))
 
     def setup_run_environment(self, env):
         """Set TCL_LIBRARY to the directory containing init.tcl.
@@ -126,7 +130,9 @@ class Tcl(AutotoolsPackage, SourceforgePackage):
 
         # WARNING: Tcl and Tcl extensions like Tk install their configuration files
         # in subdirectories like `<prefix>/lib/tcl8.6`. However, Tcl is aware of this,
-        # and $TCLLIBPATH should only contain `<prefix>/lib`.
+        # and $TCLLIBPATH should only contain `<prefix>/lib`. $TCLLIBPATH is only needed
+        # because we install Tcl extensions to different directories than Tcl. See:
+        # https://core.tcl-lang.org/tk/tktview/447bd3e4abe17452d19a80e6840dcc8a2603fcbc
         env.prepend_path(
             'TCLLIBPATH', self.spec['tcl'].libs.directories[0], separator=' ')
 

--- a/var/spack/repos/builtin/packages/tix/package.py
+++ b/var/spack/repos/builtin/packages/tix/package.py
@@ -19,8 +19,8 @@ class Tix(AutotoolsPackage):
 
     version('8.4.3', sha256='562f040ff7657e10b5cffc2c41935f1a53c6402eb3d5f3189113d734fd6c03cb')
 
-    extends('tcl')
-    depends_on('tk')
+    extends('tcl', type=('build', 'link', 'run'))
+    depends_on('tk', type=('build', 'link', 'run'))
 
     patch('https://raw.githubusercontent.com/macports/macports-ports/master/x11/tix/files/panic.patch',
           sha256='1be1a1c7453f6ab8771f90d7e7c0f8959490104752a16a8755bbb7287a841a96',
@@ -53,6 +53,12 @@ class Tix(AutotoolsPackage):
         ]
         return args
 
+    @run_after('install')
+    def darwin_fix(self):
+        # The shared library is not installed correctly on Darwin; fix this
+        if 'platform=darwin' in self.spec:
+            fix_darwin_install_name(self.prefix.lib.Tix + str(self.version))
+
     @property
     def libs(self):
         return find_libraries(['libTix{0}'.format(self.version)],
@@ -77,9 +83,3 @@ class Tix(AutotoolsPackage):
         * http://tix.sourceforge.net/docs/pdf/TixUser.pdf
         """
         env.set('TIX_LIBRARY', os.path.dirname(find(self.prefix, 'Tix.tcl')[0]))
-
-    @run_after('install')
-    def darwin_fix(self):
-        # The shared library is not installed correctly on Darwin; fix this
-        if 'platform=darwin' in self.spec:
-            fix_darwin_install_name(self.prefix.lib.Tix + str(self.version))

--- a/var/spack/repos/builtin/packages/tix/package.py
+++ b/var/spack/repos/builtin/packages/tix/package.py
@@ -25,12 +25,24 @@ class Tix(AutotoolsPackage):
     patch('https://raw.githubusercontent.com/macports/macports-ports/master/x11/tix/files/panic.patch',
           sha256='1be1a1c7453f6ab8771f90d7e7c0f8959490104752a16a8755bbb7287a841a96',
           level=0)
-    patch('https://raw.githubusercontent.com/macports/macports-ports/master/x11/tix/files/patch-generic-tixGrSort.c.diff',
-          sha256='99b33cc307f71bcf9cc6f5a44b588f22956884ce3f1e4c716ad64c79cf9c5f41',
-          level=0)
     patch('https://raw.githubusercontent.com/macports/macports-ports/master/x11/tix/files/implicit.patch',
           sha256='8a2720368c7757896814684147029d8318b9aa3b0914b3f37dd5e8a8603a61d3',
           level=0)
+    patch('https://raw.githubusercontent.com/macports/macports-ports/master/x11/tix/files/patch-generic-tixGrSort.c.diff',
+          sha256='99b33cc307f71bcf9cc6f5a44b588f22956884ce3f1e4c716ad64c79cf9c5f41',
+          level=0)
+    patch('https://raw.githubusercontent.com/macports/macports-ports/master/x11/tix/files/patch-missing-headers.diff',
+          sha256='d9f789dcfe5f4c5ee4589a18f9f410cdf162e41d35d00648c1ef37831f4a2b2b',
+          level=0)
+    patch('https://raw.githubusercontent.com/macports/macports-ports/master/x11/tix/files/patch-tk_x11.diff',
+          sha256='1e28d8eee1aaa956a00571cf495a4775e72a993958dff1cabfbc5f102e327a6f',
+          level=0)
+    patch('https://raw.githubusercontent.com/macports/macports-ports/master/x11/tix/files/patch-tk_aqua.diff',
+          sha256='41a717f5d95f61b4b8196ca6f14ece8f4764d4ba58fb2e1ae15e3240ee5ac534',
+          level=0, when='platform=darwin')
+    patch('https://raw.githubusercontent.com/macports/macports-ports/master/x11/tix/files/patch-dyld_variable.diff',
+          sha256='719eb2e4d8c5d6aae897e5f676cf5ed1a0005c1bd07fd9b18705d81a005f592b',
+          level=0, when='platform=darwin')
 
     def configure_args(self):
         spec = self.spec

--- a/var/spack/repos/builtin/packages/tix/package.py
+++ b/var/spack/repos/builtin/packages/tix/package.py
@@ -77,3 +77,9 @@ class Tix(AutotoolsPackage):
         * http://tix.sourceforge.net/docs/pdf/TixUser.pdf
         """
         env.set('TIX_LIBRARY', os.path.dirname(find(self.prefix, 'Tix.tcl')[0]))
+
+    @run_after('install')
+    def darwin_fix(self):
+        # The shared library is not installed correctly on Darwin; fix this
+        if 'platform=darwin' in self.spec:
+            fix_darwin_install_name(self.prefix.lib.Tix + str(self.version))

--- a/var/spack/repos/builtin/packages/tk/package.py
+++ b/var/spack/repos/builtin/packages/tk/package.py
@@ -25,14 +25,12 @@ class Tk(AutotoolsPackage, SourceforgePackage):
     version('8.6.3',  sha256='ba15d56ac27d8c0a7b1a983915a47e0f635199b9473cf6e10fbce1fc73fd8333')
     version('8.5.19', sha256='407af1de167477d598bd6166d84459a3bdccc2fb349360706154e646a9620ffa')
 
-    variant('xft', default=True,
-            description='Enable X FreeType')
-    variant('xss', default=True,
-            description='Enable X Screen Saver')
+    variant('xft', default=True, description='Enable X FreeType')
+    variant('xss', default=True, description='Enable X Screen Saver')
 
-    extends('tcl')
+    extends('tcl', type=('build', 'link', 'run'))
 
-    depends_on('tcl@8.6:', when='@8.6:')
+    depends_on('tcl@8.6:', type=('build', 'link', 'run'), when='@8.6:')
     depends_on('libx11')
     depends_on('libxft', when='+xft')
     depends_on('libxscrnsaver', when='+xss')
@@ -45,6 +43,21 @@ class Tk(AutotoolsPackage, SourceforgePackage):
     patch('https://github.com/macports/macports-ports/blob/master/x11/tk/files/patch-unix-Makefile.in.diff',
           sha256='54bba3d2b3550b7e2c636881c1a3acaf6e1eb743f314449a132864ff47fd0010',
           level=0, when='@:8.6.11 platform=darwin')
+    patch('https://raw.githubusercontent.com/macports/macports-ports/master/x11/tk/files/patch-dyld_fallback_library_path.diff',
+          sha256='9ce6512f1928db9987986f4d3540207c39429395d5234bd6489ba9d86a6d9c31',
+          level=0, when='platform=darwin')
+
+    def configure_args(self):
+        spec = self.spec
+        config_args = [
+            '--with-tcl={0}'.format(spec['tcl'].libs.directories[0]),
+            '--x-includes={0}'.format(spec['libx11'].headers.directories[0]),
+            '--x-libraries={0}'.format(spec['libx11'].libs.directories[0])
+        ]
+        config_args += self.enable_or_disable('xft')
+        config_args += self.enable_or_disable('xss')
+
+        return config_args
 
     def install(self, spec, prefix):
         with working_dir(self.build_directory):
@@ -63,6 +76,28 @@ class Tk(AutotoolsPackage, SourceforgePackage):
             filter_file(
                 stage_src, installed_src,
                 join_path(self.spec['tk'].libs.directories[0], 'tkConfig.sh'))
+
+    @run_after('install')
+    def symlink_wish(self):
+        with working_dir(self.prefix.bin):
+            symlink('wish{0}'.format(self.version.up_to(2)), 'wish')
+
+    def test(self):
+        self.run_test(self.spec['tk'].command.path, ['-h'],
+                      purpose='test wish command')
+
+    @property
+    def command(self):
+        """Returns the wish command.
+
+        Returns:
+            Executable: the wish command
+        """
+        # Although we symlink wishX.Y to wish, we also need to support external
+        # installations that may not have this symlink, or may have multiple versions
+        # of Tk installed in the same directory.
+        return Executable(os.path.realpath(self.prefix.bin.join(
+            'wish{0}'.format(self.version.up_to(2)))))
 
     @property
     def libs(self):
@@ -88,20 +123,3 @@ class Tk(AutotoolsPackage, SourceforgePackage):
         * https://www.tcl-lang.org/man/tcl/TkCmd/tkvars.htm
         """
         env.set('TK_LIBRARY', os.path.dirname(sorted(find(self.prefix, 'tk.tcl'))[0]))
-
-    def configure_args(self):
-        spec = self.spec
-        config_args = [
-            '--with-tcl={0}'.format(spec['tcl'].libs.directories[0]),
-            '--x-includes={0}'.format(spec['libx11'].headers.directories[0]),
-            '--x-libraries={0}'.format(spec['libx11'].libs.directories[0])
-        ]
-        config_args += self.enable_or_disable('xft')
-        config_args += self.enable_or_disable('xss')
-
-        return config_args
-
-    @run_after('install')
-    def symlink_wish(self):
-        with working_dir(self.prefix.bin):
-            symlink('wish{0}'.format(self.version.up_to(2)), 'wish')

